sdk/http-sink/src/main/java/com/google/cloud/trace/http/v1/HttpTraceSink.java

```diff
@@ -0,0 +1,67 @@
+package com.google.cloud.trace.gae.v1;
+
+import com.google.auth.oauth2.OAuth2Credentials;
+import com.google.cloud.trace.v1.sink.TraceSink;
+import com.google.common.io.ByteStreams;
+import com.google.devtools.cloudtrace.v1.Trace;
+import com.google.devtools.cloudtrace.v1.Traces;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.util.JsonFormat;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.logging.Logger;
+
+public class HttpTraceSink implements TraceSink {
+  private final static Logger logger = Logger.getLogger(HttpTraceSink.class.getName());
+
+  private final OAuth2Credentials oAuth2Credentials;
+  private final String hostUrl;
+
+  public HttpTraceSink(OAuth2Credentials oAuth2Credentials, String hostUrl) {
+    this.oAuth2Credentials = oAuth2Credentials;
+    this.hostUrl = hostUrl;
+  }
+
+  @Override
+  public void receive(Trace trace) {
+    processTraces(Traces.newBuilder().addTraces(trace).build(), trace.getProjectId());
+  }
+
+  public void processTraces(Traces traces, String projectId) {
+    URL url;
+    String urlString = String.format("%s/v1/projects/%s/traces", hostUrl, projectId);
+    try {
+      url = new URL(urlString);
+    } catch (MalformedURLException ex) {
+      logger.severe("Malformed URL: " + urlString);
+      return;
+    }
+
+    String jsonRequest;
+    try {
+      jsonRequest = JsonFormat.printer().print(traces);
+    } catch (InvalidProtocolBufferException ex) {
+      logger.severe("Failed to format: " + traces.toString());
+      return;
+    }
+
+    try {
+      HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+      connection.setRequestMethod("POST");
+      connection.setRequestProperty("X-HTTP-Method-Override", "PATCH");
+      connection.setRequestProperty("Content-Type", "application/json");
+      connection.setRequestProperty("Authorization", "Bearer " + oAuth2Credentials.getAccessToken().getTokenValue());
+      byte[] responseBytes = ByteStreams.toByteArray(connection.getInputStream());
```

Comment:
Can you move this inside the if statement to avoid making this call unless the status != 200?